server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -289,19 +292,21 @@ public void beforeCompletion() {
 
     private final Configuration configuration;
     private final SchedulerFactory schedulerFactory;
-    private final ModeManager modeManager;
+    private final CandlepinModeManager modeManager;
     private final AsyncJobStatusCurator jobCurator;
     private final OwnerCurator ownerCurator;
     private final JobMessageDispatcher dispatcher;
-    private final JobMessageSynchronizer synchronizer;
+    private final JobMessageReceiver receiver;
     private final CandlepinRequestScope candlepinRequestScope;
     private final PrincipalProvider principalProvider;
     private final Injector injector;
 
     private ManagerState state;
+    private JobMessageSynchronizer synchronizer;
     private QuartzJobExecutor qrtzExecutor;
     private Scheduler scheduler;
     private ThreadLocal<Map<String, String>> mdcState;
+    private Set<Object> suspendKeys;
```

Comment:
Somewhere (doesn't have to be this line), it would be nice to see an explanation of why suspend keys are Object, and what concrete types we might expect to see for them.

Reply:
They're like locks -- it doesn't matter what the object passed in is, so long as the same (or an equal) object is used to resume it. I use strings internally for self-documenting purposes, but that is not strictly necessary. It's just as valid if we have multiple resource endpoints that could cause jobs to be suspended, that each would pass themselves in as the "lock" to ensure that they don't unlock each other's suspend action.

I can add more to the method documentation if it doesn't adequately cover this.

Reply:
:+1: Adding a statement such as "must implement equals" to suspend keys params would work.

Reply:
But that's not strictly accurate. The only requirement would be that if equals or hashCode are overridden, they be implemented properly (or, at least, in a way that has repeatable and consistent output).

Reply:
Fair enough. The concerns I want to address are:
1) Try to avoid bugs introduced because someone passes in an instance of a custom class that *doesn't* implement `equals` properly.
2) Prevent folks from pestering you down the line with questions about what types work, etc.

It's really not a big deal though, so if you think of a line or two that helps, cool; otherwise, don't waste more time on it.